[CELEBORN-839][MR] Support Hadoop MapReduce (#1830)
apache/incubator-celeborn: README.md

Celeborn worker's slot count decreases when a partition is allocated and increments when a partition is freed.

## Build

> 1.Celeborn supports Spark 2.4/3.0/3.1/3.2/3.3/3.4 and flink 1.14/1.15/1.17 and mr hadoop-2/hadoop-3.

Reviewer: We should use the full name of the Apache Software when it is first present in the article.

2.Celeborn tested under Java 8 environment.

Build Celeborn
```shell

> ./build/make-distribution.sh -Pspark-2.4/-Pspark-3.0/-Pspark-3.1/-Pspark-3.2/-Pspark-3.3/-Pspark-3.4/-Pflink-1.14/-Pflink-1.15/-Pflink-1.17/-Phadoop-2,mr/-Pmr

Author: `-Pmr` is build flag for hadoop 3. Because we use hadoop 3 as default.

Celeborn worker's slot count decreases when a partition is allocated and increments when a partition is freed.

## Build

> 1.Celeborn supports Spark 2.4/3.0/3.1/3.2/3.3/3.4 and flink 1.14/1.15/1.17 and Hadoop 2/3.

Reviewer: ```suggestion
1.Celeborn supports Spark 2.4/3.0/3.1/3.2/3.3/3.4, Flink 1.14/1.15/1.17 and Hadoop MapReduce 2/3.
```

2.Celeborn tested under Java 8 environment.

Build Celeborn
```shell
./build/make-distribution.sh -Pspark-2.4/-Pspark-3.0/-Pspark-3.1/-Pspark-3.2/-Pspark-3.3/-Pspark-3.4/-Pflink-1.14/-Pflink-1.15/-Pflink-1.17/-Pmr
```

package apache-celeborn-${project.version}-bin.tgz will be generated.

### Package Details
Build procedure will create a compressed package.

General package layout:
```
    ├── RELEASE                         
    ├── bin                             
    ├── conf                            
    ├── jars           // common jars for master and worker                 
    ├── master-jars                     
    ├── worker-jars                     

>     ├── spark          // Spark client jars if spark profiles are activated
>     ├── flink          // flink client jars if flink profiles are activated
>     ├── mr             // mr client jars if hadoop profiles are activated

Reviewer: ```suggestion
    ├── spark          // Spark client jars if spark profiles are activated
    ├── flink          // Flink client jars if flink profiles are activated
    ├── mr             // MapReduce client jars if mr profile is activated
```

    └── flink          // flink client jars
```

MR package layout:
```
    ├── RELEASE                         
    ├── bin                             
    ├── conf                            
    ├── jars           // common jars for master and worker                        
    ├── master-jars                     
    ├── worker-jars                     
    ├── sbin

>     └── mr          // mr client jars

Reviewer: I'm wondering if we can merge the layout, and simply comment the engine's client folder as optional.

taskmanager.memory.task.off-heap.size: 512m
```

> ### Deploy mr client 

Reviewer: mr => MapReduce

Add $CELEBORN_HOME/mr/*.jar to to `mapreduce.application.classpath` and `yarn.application.classpath`.

> And setting the following settings in Yarn and MR config.

Reviewer: ```suggestion
And setting the following settings in YARN and MapReduce config.
```
